packages/salesforcedx-vscode-lightning/src/index.ts

```diff
@@ -43,17 +46,71 @@ function protocol2CodeConverter(value: string): Uri {
   return Uri.parse(value);
 }
 
+function checkActivationMode(mode: string): boolean {
+  return (
+    vscode.workspace
+      .getConfiguration('salesforcedx-vscode-lightning')
+      .get('activationMode') === mode
+  );
+}
+
 export async function activate(context: ExtensionContext) {
-  console.log('Aura Components Extension Activated');
   const extensionHRStart = process.hrtime();
+  // Run our auto detection routine before we activate
+  // 1) If activationMode is off, don't startup no matter what
+  if (checkActivationMode('off')) {
+    console.log('Aura Language Server activationMode set to off, exiting...');
+    return;
+  }
+
+  // 2) if we have no workspace folders, exit
+  if (!vscode.workspace.workspaceFolders) {
+    console.log('No workspace, exiting extension');
+    return;
+  }
+
+  // 3) If activationMode is autodetect or always, check workspaceType before startup
+  let workspaceType;
+  if (checkActivationMode('autodetect') || checkActivationMode('always')) {
+    workspaceType = lspCommon.detectWorkspaceType(
+      vscode.workspace.workspaceFolders[0].uri.fsPath
+    );
+    const sfdxWorkspace = workspaceType === WorkspaceType.SFDX;
+
+    // Check if we have a valid project structure
+    if (!lspCommon.isLWC(workspaceType)) {
```

Comment:
just add !checkActivationMode('always') here, and remove the other checks, except for line 61